code/modules/implant_hacking/_hacking.dm

```diff
@@ -0,0 +1,162 @@
+/obj/item/cyberlink_connector
+	name = "Wireless personal connector"
+	desc = "Allows you to connect to incompatible implants and hack them."
+	icon = 'icons/obj/surgery.dmi'
+	icon_state = "connector"
+	var/obj/item/organ/cyberimp/cyberlink/parent_cyberlink
+	var/list/datum/hacking_minigame/game_list = list()
+	var/time_left = 0
+	var/obj/item/organ/cyberimp/cybernetic
+	var/mob/living/current_user
+	var/processing = FALSE
+
+/obj/item/cyberlink_connector/Destroy()
+	. = ..()
+	parent_cyberlink = null
+	cleanup()
+	STOP_PROCESSING(SSprocessing,src)
+	processing = FALSE
+
+///We dont open the tgui when we click on this.
+/obj/item/cyberlink_connector/interact(mob/user)
+	add_fingerprint(user)
+
+/obj/item/cyberlink_connector/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+	. = ..()
+	if(!proximity_flag)
+		return
+
+	if(!istype(target,/obj/item/organ/cyberimp) || istype(target,/obj/item/organ/cyberimp/cyberlink))
+		return
+
+	if(!parent_cyberlink)
+		var/obj/item/organ/cyberimp/cyberlink/link = user.getorganslot(ORGAN_SLOT_LINK)
+		if(!link)
+			to_chat(user,"<span class='notice'> NO CYBERLINK DETECTED </span>")
+			return
+		parent_cyberlink = link
+
+	game_list = list()
+
+	var/diffrences = 3
+	current_user = user
+	cybernetic = target
+
+	for(var/info in cybernetic.encode_info)
+		if(cybernetic.encode_info[info] == 0)
+			continue
+		var/list/encrypted_information = cybernetic.encode_info[info]
+
+		for(var/protocol in encrypted_information)
+			if(protocol in parent_cyberlink.encode_info[info])
+				diffrences--
+				break
+
+	if(diffrences == 0)
+		to_chat(current_user,"<span class='notice'> Cyberlink beeps: [uppertext(cybernetic.name)] ALREADY COMPATIBLE.</span>")
+		cleanup()
+		return
+
+	if(!game_list.len)
```

Comment:
When is this ever not the case?

Reply:
It is not the case when the game has already started and someone decided to close the TGUI, the game is not supposed to stop then, but rather finish in the background, and if someone decides to open the TGUI while the game is still in progress, it will not generate a new board but instead display the previous board with appropriate amount of time left.